eth_crowdfund_api/src/views/CampaignView.py

```diff
@@ -1,61 +1,62 @@
 from flask import request, json, Response, Blueprint, jsonify
 from ..models.Campaign import Campaign, CampaignSchema
-
+from marshmallow import ValidationError
+from sqlalchemy import exc
+from . import custom_response
 
 
 campaign_api = Blueprint('campaigns_api', __name__)
 campaign_schema = CampaignSchema()
 
 @campaign_api.route('/', methods=['POST'])
 def create():
-  req_data = request.get_json()
-  data = campaign_schema.load(req_data)
-  # data, error = campaign_schema.load(req_data)
-  # BUT WITH ERROR: the above code threw an error, currently ignoring errors
-  error = None
-  if error: 
-    return custom_response(error, 400)
+  try: 
+    req_data = request.get_json()
+    data = campaign_schema.load(req_data)
+  except ValidationError as err: 
+    return custom_response(err.messages, 400)
+  except: 
+    return custom_response({ "error": "No input data provided" }, 400)
 
   campaign = Campaign(data)
   campaign.save() 
   campaign_data = campaign_schema.dump(campaign)
   return custom_response(campaign_data, 201)
-  
-
-def custom_response(res, status_code):
-  return Response(
-    mimetype="application/json",
-    response=json.dumps(res),
-    status=status_code
-  )
 
 @campaign_api.route('/<int:campaign_id>', methods=['GET'])
 def get_a_campaign(campaign_id):
   campaign = Campaign.get_one_campaign(campaign_id)
-  if not campaign: 
+  if campaign is None: 
     return custom_response({'error':'Campaign not found'}, 404)
 
   campaign_data = campaign_schema.dump(campaign)
   return custom_response(campaign_data, 200)
 
 @campaign_api.route('/<int:campaign_id>', methods=['PUT'])
 def update(campaign_id):
-  req_data = request.get_json()
-  data = campaign_schema.load(req_data, partial=True)
-  # error not functional yet
-  error = None
-  if error:
-    return custom_response(error, 400)
-  
-  campaign = Campaign.get_one_campaign(campaign_id)
-  campaign.update(data)
+  try: 
+    req_data = request.get_json()
+    data = campaign_schema.load(req_data, partial=True)
+    campaign = Campaign.get_one_campaign(campaign_id)
+    campaign.update(data)
+  except ValidationError as err: 
+    return custom_response(err.messages, 400)
+  except exc.IntegrityError as err: 
+    return custom_response({ "error": err.orig.diag.message_detail}, 400)
+  except: 
+    return custom_response({ "error": "No input data provided" }, 400)
```

Comment:
This will catch all other errors and not just errors for no input data provided. Possible refactor opportunity